cc-statistics/README.md

```diff
@@ -0,0 +1,236 @@
+
+### Requirements
+
+Python (2.x)
+
+### Quick Start
+
+
+#### Rebuilding String maps
+In order to get pyWebGraph working we need to (rebuild the string maps)[http://law.di.unimi.it/tutorial.php].
+Following the above steps to get the notebook started, you should have a vertices file (cc-main-2020-feb-mar-may-domain-vertices-domain.txt.gz) which contains a mapping from a node id to a reversed domain name. We however need to attain a mapping from reversed domain name to numbers. To achieve this we will use the (minimal perfect hashing algorithm)[http://sux4j.di.unimi.it/docs/it/unimi/dsi/sux4j/mph/package-summary.html]. 
+
+Running the following command will produce a `mph` (minimal perfect hash) file that contains mapping from reversed domain name to number.
+
+```
+zcat cc-main-2020-feb-mar-may-domain-vertices.txt.gz \
+  | cut -f2 \
+  | $WG it.unimi.dsi.sux4j.mph.GOV4Function cc-main-2020-feb-mar-may-domain.mph
+``` 
+
+However, we would also like to have the functionality of determining if a string is actually present as a domain in the map. We will produce a `String Map` to do so.
+
+Running the following command will produce a `smph` file (string map perfect hash) that has the outlined functionality.
+
+```
+zcat cc-main-2020-feb-mar-may-domain-vertices.txt.gz \
+  | cut -f2 \
+  | $WG it.unimi.dsi.util.ShiftAddXorSignedStringMap cc-main-2020-feb-mar-may-domain.mph cc-main-2020-feb-mar-may-domain.smph
+```
+
+We will also need a front coded list to map node numbers to URLs:
+
+```
+zcat cc-main-2020-feb-mar-may-domain-vertices.txt.gz \
+  | cut -f2 \
+  | $WG it.unimi.dsi.util.FrontCodedStringList -u -r 32 cc-main-2020-feb-mar-may-domain.fcl
+```
+
+Running these three commands, one should obtain an `mph` file, `smph` file and a `fcl` file. We may now begin the installation process for pyWebGraph!
+
+#### Jython and pyWebGraph
+
+We will clone the pyWebGraph repo:
+
+`git clone https://github.com/mapio/py-web-graph.git`
+
+We will now operate in this repo:
+
+`cd py-web-graph`
+
+We require Jython to run pyWebGraph, so we will install it: (more info at https://www.jython.org/download)
+
+`wget https://repo1.maven.org/maven2/org/python/jython-standalone/2.7.2/jython-standalone-2.7.2.jar`
+
+We will copy the console to the root so `pywebgraph` may be visible as a package.
+
+`cp pywebgraph/console.py .`
+
+Now we define shell variable `DIR` to be the location of where WebGraph is installed. Assuming it is in the current working directory:
+
+`DIR=$PWD`
+
+Defining the Web Graph version (which is visible by the directory name of webgraph):
+
+`WEBGRAPH_VERSION="3.6.5"`
+
+We will now define the class path for the webgraph package by running:
+
+`WG_CP=$DIR/webgraph-$WEBGRAPH_VERSION.jar:$(ls $DIR/deps/*.jar | tr '\n' ':')`
+
+Note for Windows users, the seperator for class paths is `;` rather than `:`. This means that this command will become:
+
+`WG_CP=$DIR/webgraph-$WEBGRAPH_VERSION.jar\;$(ls $DIR/deps/*.jar | tr '\n' ';')`
```

Comment:
If adding WebGraph and LAW in the assembly jar works, `WG_CP` would only contain `.../cc-webgraph/target/cc-webgraph-0.1-SNAPSHOT-jar-with-dependencies.jar`